libavformat/hls.c

```diff
@@ -2789,6 +2789,7 @@ static int hls_read_seek(AVFormatContext *s, int stream_index,
 
         pls->seek_timestamp = seek_timestamp;
         pls->seek_flags = flags;
+        pls->open_next_segment = 1;
```

Comment:
How does this fix the problem? How did it break in the first place?

Reply:
The intention of that first commit was to help make sure we flushed the remaning packets in the  current segment when it ended, instead of waiting for the start of the next segment. 

This was causing us to have to skip or wait SEGMENT_DURATION to get all the scores for a segment.